src/utils/handleRequest.ts

```diff
@@ -65,11 +66,20 @@ export async function handleRequest<
   }
 
   // Resolve a mocked response from the list of request handlers.
-  const lookupResult = await getResponse(
-    request,
-    handlers,
-    handleRequestOptions?.resolutionContext,
-  )
+  const [lookupError, lookupResult] = await until(() => {
+    return getResponse(
+      request,
+      handlers,
+      handleRequestOptions?.resolutionContext,
+    )
+  })
+
+  if (lookupError) {
+    // Allow developers to react to unhandled exceptions in request handlers.
+    emitter.emit('unhandledException', lookupError, request)
```

Comment:
I'm using the `unhandledException` event name (dropping the "request:" part) because exceptions are not related to requests being handled. They are related to request handlers or resolvers. I think a context-free `unhandledException` makes more sense in that regard.